src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -0,0 +1,485 @@
+            {
+                const string DiagnosticOptionPrefix = "dotnet_diagnostic.";
+                const string DiagnosticOptionSuffix = ".severity";
+
+                foreach (var (key, value) in section.Properties)
+                {
+                    // Keys are lowercased in editorconfig parsing
+                    if (key.StartsWith(DiagnosticOptionPrefix, StringComparison.Ordinal) &&
+                        key.EndsWith(DiagnosticOptionSuffix, StringComparison.Ordinal))
+                    {
+                        var diagId = key.Substring(
+                            DiagnosticOptionPrefix.Length,
+                            key.Length - (DiagnosticOptionPrefix.Length + DiagnosticOptionSuffix.Length));
+
+                        ReportDiagnostic? severity;
+                        var comparer = StringComparer.OrdinalIgnoreCase;
+                        if (comparer.Equals(value, "default"))
+                        {
+                            severity = ReportDiagnostic.Default;
+                        }
+                        else if (comparer.Equals(value, "error"))
+                        {
+                            severity = ReportDiagnostic.Error;
+                        }
+                        else if (comparer.Equals(value, "warn"))
+                        {
+                            severity = ReportDiagnostic.Warn;
+                        }
+                        else if (comparer.Equals(value, "info"))
+                        {
+                            severity = ReportDiagnostic.Info;
+                        }
+                        else if (comparer.Equals(value, "hidden"))
+                        {
+                            severity = ReportDiagnostic.Hidden;
+                        }
+                        else if (comparer.Equals(value, "suppress"))
+                        {
+                            severity = ReportDiagnostic.Suppress;
+                        }
+                        else
+                        {
+                            severity = null;
+                            diagnosticBuilder.Add(Diagnostic.Create(
+                                InvalidAnalyzerConfigSeverityDescriptor,
+                                Location.None,
+                                diagId,
+                                value,
+                                analyzerConfigPath));
+                        }
+
+                        if (severity.HasValue)
+                        {
+                            treeBuilder[diagId] = severity.GetValueOrDefault();
+                        }
+                    }
+                    else
+                    {
+                        analyzerBuilder[key] = value;
+                    }
+                }
+            }
+        }
+
+        private static DiagnosticDescriptor InvalidAnalyzerConfigSeverityDescriptor
+            = new DiagnosticDescriptor(
+                "InvalidSeverityInAnalyzerConfig",
```

Comment:
>InvalidSeverityInAnalyzerConfig [](start = 17, length = 31)

Is this the id presented to the user? If so, should it be of the form "LLNNNN"?

Reply:
Do we have other instances of diagnostic ids that are not two letters, 4 digits?

---
In reply to: [228266291](https://github.com/dotnet/roslyn/pull/30752#discussion_r228266291) [](ancestors = 228266291)

Reply:
No, but because this message isn't tied to a specific language, I don't know what structure to use. In this case I thought deviating from the standard is actually *good* because it doesn't indicate that this is a C# or VB-specific error (it's just a general editorconfig error).